Shared/Samples/Filter features in scene/FilterFeaturesInSceneView.swift

```diff
@@ -0,0 +1,241 @@
+    /// The view model for this sample.
+    @StateObject private var model = Model()
+    
+    /// The filter state for the scene view.
+    @State private var filterState: FilterState = .load
+    
+    /// A Boolean value indicating whether to show an error alert.
+    @State private var isShowingAlert = false
+    
+    /// The error shown in the error alert.
+    @State var error: Error? {
+        didSet { isShowingAlert = error != nil }
+    }
+    
+    var body: some View {
+        SceneView(scene: model.scene, graphicsOverlays: [model.graphicsOverlay])
+            .task {
+                do {
+                    try await model.scene.load()
+                } catch {
+                    self.error = error
+                }
+            }
+            .toolbar {
+                ToolbarItemGroup(placement: .bottomBar) {
+                    Button(filterState.label) {
+                        model.handleFilterState(filterState)
+                        filterState = filterState.next()
+                    }
+                }
+            }
+            .alert(isPresented: $isShowingAlert, presentingError: error)
+    }
+    
+    /// The different states for filtering features in a scene.
+    enum FilterState: Equatable {
+        case load, filter, reset
+        
+        /// A human-readable label for the filter state.
+        var label: String {
+            switch self {
+            case .load: return "Load"
+            case .filter: return "Filter"
+            case .reset: return "Reset"
+            }
+        }
+        
+        /// The next filter state to apply to a scene.
+        func next() -> Self {
+            switch self {
+            case .load:
+                return .filter
+            case .filter:
+                return .reset
+            case .reset:
+                return .load
+            }
+        }
+    }
+}
+
+private extension FilterFeaturesInSceneView {
+    /// The model used to store the geo model and other expensive objects
+    /// used in this view.
+    class Model: ObservableObject {
+        /// The scene for this sample.
+        let scene: ArcGIS.Scene
+        
+        /// The open street map layer for the sample.
+        private let osmBuildings = ArcGISSceneLayer(
+            item: PortalItem(
+                portal: .arcGISOnline(connection: .anonymous),
+                id: .osmBuildings
+            )
+        )
+        
+        /// The San Francisco buildings scene layer for the sample.
+        private let buildingsSceneLayer = ArcGISSceneLayer(
+            url: URL(string: "https://tiles.arcgis.com/tiles/z2tnIkrLQ2BRzr6P/arcgis/rest/services/SanFrancisco_Bldgs/SceneServer")!
+        )
+        
+        /// The graphics overlay for the scene view.
+        let graphicsOverlay = GraphicsOverlay()
+        
+        /// A polygon that shows the extent of the detailed buildings scene layer.
+        private let polygon: ArcGIS.Polygon = makeFilteringPolygon()
```

Comment:
> 2. So we create a method stub that produces a polygon to simulate this data source.

Are you suggesting to create a method that takes an extent as a parameter and returns a polygon with the red line graphic applied? 

> 4. If the geometry of the graphic changes, the new polygon can be applied.

Should these changes be state driven? Such as a state variable for the extent changes and that updates the polygon and red graphic?

Reply:
> Are you suggesting to create a method that takes an extent as a parameter and returns a polygon with the red line graphic applied?

This makes me rethink my comment 🤔 . How can user get a polygon in a scene view? Cannot be from geometry editor which is 2D only. Probably either from a feature's geometry, or from a list of prebaked polygons. In either case I think a static method that returns a polygon makes sense.

In terms of the graphic, I don't have strong opinion. It can either be an output of a method, or a constant with a changeable geometry property. The graphic re-render is handled by the core so it doesn't need to be a state property, can just be a variable in the data model.

Reply:
Okay I will leave the static polygon method as-is, as well as the computed property for the red graphic. Since the graphic uses the polygon initialized with a static method, the graphic unfortunately cannot be a constant.

Reply:
```swift
private let sanFranciscoExtentGraphic: Graphic

init() {
    // …
    sanFranciscoExtentGraphic = Graphic(
        geometry: polygon,
        symbol: simpleFillSymbol
    )
}
```